### DISCUSSION TITLE
how to register users when EMAIL_PROVIDER=NULL ?

### DISCUSSION THREAD

**Discussion by ricardogsilva** [fief-dev/fief#311] [Q&A] [answered]
Hi

The [docs say that the default email provider is `EMAIL_PROVIDER=NULL`](https://docs.fief.dev/self-hosting/deployment/setup-email-provider/#null-provider). 

However, when I try to register a new user, fief tells me that it has sent an email with a verification code and that I must enter it in the form. Having not configured any mail provider (and therefore using the default `EMAIL_PROVIDER=NULL`) I am wondering how I would be able to bypass this? 

Does this mean that it is indeed mandatory to configure a mail provider in order to let users self-register with fief? Is this also mandatory during development? 

If it is mandatory, then I think the docs should be more explicit about it - Specifically for dev purposes, I'm now looking for some email server that I can set up easily just for this - but it feels like I'm jumping off to a side quest here, deviating from my main goal, which would be to get fief auth integrated and up and running with my web app, which is what I actually want to spend my time developing.

Additionally, I'm wondering if it makes sense to even have a null email provider, if that makes fief self-registration unusable.

P.S. Just to mention that I am trying out the self-hosted version of fief

**Comment by frankie567** [comment, marked as answer]
Hi @ricardogsilva 👋 

It's true that this is case is not really well documented. Actually, when the null email provider is used, a `WARNING` log is displayed with the verification code, so you can just copy/paste it directly.

You can see the logs with the following command (assuming your Fief container is named `fief-server`):

```sh
docker logs -f fief-server
```

---

Admittedly, this should probably be mentioned in the documentation. Besides, we are working on disabling the mandatory email verification, so it'll be possible to disable this during development.
